service/src/main/java/datawave/microservice/query/executor/action/ExecutorTask.java

```diff
@@ -299,77 +298,91 @@ protected RESULTS_ACTION shouldGenerateMoreResults(boolean exhaust, TaskKey task
         }
     }
     
-    protected boolean pullResults(QueryLogic queryLogic, CachedQueryStatus queryStatus, boolean exhaustIterator) throws Exception {
-        // start the timer on the query status to ensure we flush numResultsGenerated updates periodically
-        queryStatus.startTimer();
-        
+    protected boolean pullResults(QueryLogic queryLogic, Query query, boolean exhaustIterator) throws Exception {
         if (queryLogic instanceof CheckpointableQueryLogic && ((CheckpointableQueryLogic) queryLogic).isCheckpointable()) {
             queryTaskUpdater.setQueryLogic((CheckpointableQueryLogic) queryLogic);
         }
-        try {
-            TaskKey taskKey = task.getTaskKey();
-            String queryId = taskKey.getQueryId();
-            TransformIterator iter = queryLogic.getTransformIterator(queryStatus.getQuery());
-            long maxResults = queryLogic.getResultLimit(queryStatus.getQuery());
-            if (maxResults != queryLogic.getMaxResults()) {
-                log.info("Maximum results set to " + maxResults + " instead of default " + queryLogic.getMaxResults() + ", user "
-                                + queryStatus.getQuery().getUserDN() + " has a DN configured with a different limit");
-            }
-            if (queryStatus.getQuery().isMaxResultsOverridden()) {
-                maxResults = Math.max(maxResults, queryStatus.getQuery().getMaxResultsOverride());
-            }
-            int pageSize = queryStatus.getQuery().getPagesize();
-            if (queryLogic.getMaxPageSize() != 0) {
-                pageSize = Math.min(pageSize, queryLogic.getMaxPageSize());
-            }
-            QueryResultsPublisher publisher = resultsManager.createPublisher(queryId);
-            RESULTS_ACTION running = shouldGenerateMoreResults(exhaustIterator, taskKey, pageSize, maxResults, queryStatus);
-            int count = 0;
-            while (running == RESULTS_ACTION.GENERATE && iter.hasNext()) {
-                count++;
-                try {
-                    Object result = iter.next();
+        
+        TaskKey taskKey = task.getTaskKey();
+        String queryId = taskKey.getQueryId();
+        TransformIterator iter = queryLogic.getTransformIterator(query);
+        long maxResults = queryLogic.getResultLimit(query);
+        if (maxResults != queryLogic.getMaxResults()) {
+            log.info("Maximum results set to " + maxResults + " instead of default " + queryLogic.getMaxResults() + ", user " + query.getUserDN()
+                            + " has a DN configured with a different limit");
+        }
+        if (query.isMaxResultsOverridden()) {
+            maxResults = Math.max(maxResults, query.getMaxResultsOverride());
+        }
+        int pageSize = query.getPagesize();
+        if (queryLogic.getMaxPageSize() != 0) {
+            pageSize = Math.min(pageSize, queryLogic.getMaxPageSize());
```

Comment:
I dont know what we do now, but should we be silently lowering this here rather than throwing an exception. This seems like a malformed query to me.

Reply:
I'm not sure why this code is here.  The QueryManagementService runs before this, and throws an exception if the query logic has a max page size, and the user's page size exceeds that.  Here's the code:

```
        // always check against the max
        QueryParameters queryParameters = getQueryParameters();
        if (queryLogic.getMaxPageSize() > 0 && queryParameters.getPagesize() > queryLogic.getMaxPageSize()) {
            log.error("Invalid page size: {} vs {}", queryParameters.getPagesize(), queryLogic.getMaxPageSize());
            throw new BadRequestQueryException(DatawaveErrorCode.PAGE_SIZE_TOO_LARGE, MessageFormat.format("Max = {0}.", queryLogic.getMaxPageSize()));
        }
```

@ivakegg can we get rid of this in the executor task?

Reply:
I think in the webservice we currently throw an exception which the QueryManagerService is consistent with.  I should either not check here or also throw and exception incase something slips through I guess.

Reply:
it really does not matter so much here however.  the page size is only used to determine whether additional results should be added to the queue.  The management service is really the only one controlling the actual page size returned.  I am leaning towards removing the check.